<launch>
  <arg name="b_control_joy_topic" default="/b_control/joy" />
  <arg name="spacenav_joy_topic" default="/spacenav/joy" />
  <arg name="default_frame_id" default="odom_on_ground" />
  <arg name="use_icp" default="true" />
  <!-- robot specific setting -->
  <arg name="trackball_head_launch" default="hrp2_trackball_head.launch" />
  <arg name="enable_trackball_head" default="true" />
  
  <!-- publish spacenav topic -->
  <node pkg="spacenav_node" type="spacenav_node" name="spacenav">
  </node>

  <!-- publish midi device topic -->
  <node pkg="jsk_teleop_joy" type="midi_config_player.py" name="midi_config_player"
        args="$(find jsk_teleop_joy)/configs/b_control.yaml"
        output="screen">
    <remap from="joy" to="b_control/joy" />
    <param name="autorepeat_rate" value="0" />
  </node>

  <!-- b_control client node -->
  <node pkg="drc_task_common" type="b_control_client.py" name="b_control_client"
        output="screen">
    <remap from="input_joy" to="$(arg b_control_joy_topic)"/>
    <remap if="$(arg use_icp)" from="selected_box" to="/passed_selected_box" />
    <remap unless="$(arg use_icp)" from="selected_box" to="/bounding_box_marker/selected_box" />
    <remap if="$(arg use_icp)" from="t_marker_info" to="/t_marker_information" />
    <!-- range of marker dimensions -->
    <param name="transformable_interactive_server_nodename" value="transformable_interactive_server"/>
    <param name="default_frame_id" value="$(arg default_frame_id)"/>
    <param name="x_max" value="1.5"/>
    <param name="y_max" value="1.5"/>
    <param name="z_max" value="1.5"/>
    <param name="r_max" value="1.0"/>
    <param name="sr_max" value="0.2"/>
    <param name="x_min" value="0.01"/>
    <param name="y_min" value="0.01"/>
    <param name="z_min" value="0.01"/>
    <param name="r_min" value="0.01"/>
    <param name="sr_min" value="0.001"/>
  </node>
  
  <!-- spacenav client node -->
  <node pkg="drc_task_common" type="spacenav_client.py" name="spacenav_client">
    <remap from="input_joy" to="$(arg spacenav_joy_topic)"/>
    <!-- choose using both at same time or using separately -->
    <param name="transformable_interactive_server_nodename" value="transformable_interactive_server"/>
    <param name="separate_mode" value="true"/>
    <!-- these are diff speed gain -->
    <param name="x_max" value="0.01"/>
    <param name="y_max" value="0.01"/>
    <param name="z_max" value="0.01"/>
    <param name="rx_max" value="0.01"/>
    <param name="ry_max" value="0.01"/>
    <param name="rz_max" value="0.01"/>
  </node>

  <!-- trackball client node -->
  <include if="$(arg enable_trackball_head)" file="$(find jsk_teleop_joy)/launch/$(arg trackball_head_launch)">
  </include>
  <include file="$(find drc_task_common)/launch/joy_teleop.launch">
    <arg name="launch_joy" default="false" />
  </include>
</launch>
